StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1128ConstructorInitializerMustBeOnOwnLine.cs

```diff
@@ -0,0 +1,80 @@
+    public class SA1128ConstructorInitializerMustBeOnOwnLine : DiagnosticAnalyzer
+    {
+        /// <summary>
+        /// The ID for diagnostics produced by the <see cref="SA1128ConstructorInitializerMustBeOnOwnLine"/>
+        /// </summary>
+        public const string DiagnosticId = "SA1128";
+        private static readonly LocalizableString Title = new LocalizableResourceString(nameof(ReadabilityResources.SA1128Title), ReadabilityResources.ResourceManager, typeof(ReadabilityResources));
+        private static readonly LocalizableString MessageFormat = new LocalizableResourceString(nameof(ReadabilityResources.SA1128MessageFormat), ReadabilityResources.ResourceManager, typeof(ReadabilityResources));
+        private static readonly LocalizableString Description = new LocalizableResourceString(nameof(ReadabilityResources.SA1128Description), ReadabilityResources.ResourceManager, typeof(ReadabilityResources));
+        private static readonly string HelpLink = "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1128.md";
+
+        private static readonly DiagnosticDescriptor Descriptor =
+            new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, AnalyzerCategory.ReadabilityRules, DiagnosticSeverity.Warning, AnalyzerConstants.EnabledByDefault, Description, HelpLink);
+
+        private static readonly ImmutableArray<DiagnosticDescriptor> SupportedDiagnosticsValue =
+            ImmutableArray.Create(Descriptor);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        {
+            get
+            {
+                return SupportedDiagnosticsValue;
+            }
+        }
+
+        /// <inheritdoc/>
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(HandleCompilationStart);
+        }
+
+        private static void HandleCompilationStart(CompilationStartAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleConstructor, SyntaxKind.ConstructorDeclaration);
+        }
+
+        private static void HandleConstructor(SyntaxNodeAnalysisContext context)
+        {
+            var constructor = (ConstructorDeclarationSyntax)context.Node;
+            if (constructor.Initializer != null)
+            {
+                Analyze(context, constructor);
+            }
+        }
+
+        private static void Analyze(
+            SyntaxNodeAnalysisContext context,
+            ConstructorDeclarationSyntax constructor)
+        {
+            var initializer = constructor.Initializer;
+            var colon = initializer.ColonToken;
+
+            if (!colon.IsFirstInLine())
+            {
+                context.ReportDiagnostic(Diagnostic.Create(Descriptor, initializer.GetLocation()));
+                return;
+            }
+
+            if (colon.TrailingTrivia.Any(t => t.IsKind(SyntaxKind.EndOfLineTrivia)))
```

Comment:
:bulb: `.Any(SyntaxKind.EndOfLineTrivia)` is a more efficient overload (fixed during merge).